test/trusted_resources_test.go

```diff
@@ -0,0 +1,315 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	"github.com/tektoncd/pipeline/pkg/pod"
+	"github.com/tektoncd/pipeline/test/parse"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/system"
+	knativetest "knative.dev/pkg/test"
+	"knative.dev/pkg/test/helpers"
+)
+
+var (
+	neededFeatureFlags = map[string]string{
+		"resource-verification-mode": "enforce",
+		// Make sure this is running under alpha integration tests
+		"enable-api-fields": "alpha",
+	}
+	privKey  = "trustedresources-keys/cosign.key"
+	pubKey   = "trustedresources-keys/cosign.pub"
+	password = "1234"
+)
+
+func init() {
+	// This is the password for test/trustedresources-keys/cosign.key, only used in this test.
+	os.Setenv("PRIVATE_PASSWORD", password)
+}
+
+func TestTrustedResourcesVerify_Success(t *testing.T) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	c, namespace := setup(ctx, t, requireAnyGate(neededFeatureFlags))
+
+	// Note that this may not work if we run e2e tests in parallel since this feature flag require all tasks and pipelines
+	// to be signed and unsigned resources will fail.
+	configMapData := map[string]string{
+		"resource-verification-mode": config.EnforceResourceVerificationMode,
+	}
+	if err := updateConfigMap(ctx, c.KubeClient, system.Namespace(), config.GetFeatureFlagsConfigName(), configMapData); err != nil {
+		t.Fatal(err)
+	}
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(ctx, t, c, namespace) }, t.Logf)
+	defer tearDown(ctx, t, c, namespace)
```

Comment:
just triple checking - will `resource-verification-mode` be reset back to it's original value after the test even if the test is interrupted or fails? (i.e. does teardown reset the value?)

Reply:
Thanks! I didn't put the cleanup code inside the tearDown, I feel like tearDown is mainly used to clean the namespace we created for this test. Our current test need to reset the configmap at system namespace, so to address your comment I create a new helperfunction within this test to clean up the created secret and configmap.

The defer will enable the function to be executed even if the test fails. I have tested this behaviour. 

Thank you very much for pointing this out!!